src/ui/components/Library/Team/View/TestRuns/TestRunsContextRoot.tsx

```diff
@@ -40,23 +39,21 @@ type TestRunsContextType = {
 export const TestRunsContext = createContext<TestRunsContextType>(null as any);
 
 export function TestRunsContextRoot({ children }: { children: ReactNode }) {
-  const { teamId, testOrTestRunId: defaultTestRunId } = useGetTeamRouteParams();
+  const { teamId, testOrTestRunId, testId } = useGetTeamRouteParams();
 
   const { testRuns, status } = useTestRuns();
 
-  const [testRunId, setTestRunId] = useState<string | null>(defaultTestRunId);
-
+  const [localTestRunId, setTestRunId] = useState<string | null>(testOrTestRunId);
```

Comment:
> When I passed it the value from the URL, the URL and deferred value were always the same.

That's because `useDeferredValue` will be the same as whatever value you pass it initially (else it wouldn't work, from a typing perspective), and changes to URL values don't schedule updates with React (unless something is listening and telling React that it needs to update)– which could be done by mirroring the value in state, or you could use the sync external store hook to subscribe to URL changes.

That being said, it's not clear to me what you're trying to do pulling a value from a URL and why you would want/need that to be "deferred". Deferred values are values that get updated as part of a "transition"– which only matters for updates, not mounts. URL values usually only apply to mounts.

Reply:
That makes sense, thanks! I haven't worked with suspense enough yet to have learned all of this so I appreciate the info.

I don't think we need the deferred values except that the loading states in the views rely on the difference between the displayed and actual values. That'll need to be handled differently.

Reply:
Maybe you and I can chat about this in realtime sometime this week? Could be easier.